src/rt/GEAR/rt_grackle_cooling.h

```diff
@@ -0,0 +1,228 @@
+/*******************************************************************************
+ * This file is part of SWIFT.
+ * Copyright (c) 2020 Mladen Ivkovic (mladen.ivkovic@hotmail.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ ******************************************************************************/
+#ifndef SWIFT_RT_GEAR_GRACKLE_COOLING_H
+#define SWIFT_RT_GEAR_GRACKLE_COOLING_H
+
+#include "chemistry.h"
+
+/**
+ * @file src/rt/GEAR/rt_grackle_cooling.h
+ * @brief header file for the GEAR M1 closure radiative transfer scheme
+ * kiara grackle cooling related functions.
+ */
+
+/**
+ * @brief Returns the value of G0 for given particle p
+ *
+ * @param p Pointer to the particle data.
+ * @param cooling The properties of the cooling function.
+ *
+ */
+__attribute__((always_inline)) INLINE float cooling_compute_G0(
+                const struct part *restrict p,
+                const struct rt_props* rt_props) {
+
+      float G0 = 0.f;
+      /* Determine ISRF in Habing units based on chosen method */
+      if (rt_props->G0_computation_method==0) {
+          G0 = 0.f;
+      }
+      else if (rt_props->G0_computation_method==1) {
+          G0 = p->chemistry_data.local_sfr_density * rt_props->G0_factor1;
+      }
+      else if (rt_props->G0_computation_method==2) {
+          G0 = p->group_data.ssfr * rt_props->G0_factor2;
+      }
+      else if (rt_props->G0_computation_method==3) {
+          if (p->group_data.ssfr > 0.) {
+              G0 = p->group_data.ssfr * rt_props->G0_factor2;
+          }
+          else {
+              G0 = p->chemistry_data.local_sfr_density * rt_props->G0_factor1;
+          }
+      }
+      return G0;
+}
+
+/**
+ * @brief get the densities of all species and electrons.
+ *
+ * @param p particle to use
+ * @param rho particle physical density
+ * @param species_densities array to write densities in
+ *
+ **/
+__attribute__((always_inline)) INLINE static void
+rt_tchem_get_species_densities(const struct part* restrict p, gr_float rho,
```

Comment:
Since I need to use the struct rt_props where I cannot use it in the rt_thermochemistry_utils.h. So I copy this function and put it into the new header file rt_grackle_cooling.h. We discuss this issue before